### PR TITLE
fix(jiva-csi): volume mounted at two places

### PR DESCRIPTION
**Pull request by utkarshmani1997**
- NodePublishVolume was getting triggered before NodeUnPublishVolume
  RPC call due to which volume is seen mounted at two places.
- Above case is true even when NodeStageVolume RPC is call is triggered
  before NodeUnPublishVolume.

Signed-off-by: Utkarsh Mani Tripathi <utkarsh.tripathi@mayadata.io>

### CODE REVIEW

File: pkg/driver/node.go [hunk 1]
```diff
@@ -247,6 +248,11 @@ func (ns *node) NodeStageVolume(
 		return nil, status.Error(codes.FailedPrecondition, err.Error())
 	}
 
+	// Volume may be mounted at targetPath (bind mount in NodePublish)
+	if err := ns.isAlreadyMounted(reqParam.volumeID); err != nil {
```
Comment (payes) [inline pkg/driver/node.go]: This check will remove the idempotent behavior of NodeStageVolume.

**Review by payes — APPROVED** [review]

File: pkg/driver/node.go [hunk 2]
```diff
@@ -512,17 +545,62 @@ func (ns *node) NodeUnpublishVolume(
...
+		if strings.Contains(mntInfo.Path, volID) {
+			currentMounts[mntInfo.Path] = true
+		}
+	}
+
+	// if volume is mounted at more than one place check if this request is
+	// for the same path that is already mounted. Return nil if the path is
+	// mounted already else return err so that it gets unmounted in the
+	// next subsequent calls in respective rpc calls (NodeUnpublishVolume, NodeUnstageVolume)
+	if len(currentMounts) > 1 {
```
Comment (pawanpraka1) [inline pkg/driver/node.go:582]: even if it is mounted at one place, shouldn't we verify in that case also that it is mounted at the corresponding targetpath or not? it should len(currentMounts) > 0?

Comment (pawanpraka1) [reply pkg/driver/node.go:582]: can you test a Deployment with two replica and see if it is working.

Comment (utkarshmani1997) [reply pkg/driver/node.go:582]: yes, you are right but we are not supporting RWM.

Comment (pawanpraka1) [reply pkg/driver/node.go:582]: please make a note of it that we don't allow to mount a pvc more than once even on the same node.

**Review by pawanpraka1 — APPROVED** [review]
looks good.

**Pull request merged**